pandas/tests/io/test_sql.py

```diff
@@ -1430,12 +1771,26 @@ def test_api_custom_dateparsing_error(
             "DateCol": {"errors": error},
         },
     )
+
     if "postgres" in conn_name:
         # TODO: clean up types_data_frame fixture
         result = result.drop(columns=["DateColWithTz"])
         result["BoolCol"] = result["BoolCol"].astype(int)
         result["BoolColWithNull"] = result["BoolColWithNull"].astype(float)
 
+    if conn_name == "postgresql_adbc_conn":
+        expected = expected.astype(
+            {
+                "IntDateCol": "int32",
+                "IntDateOnlyCol": "int32",
+                "IntCol": "int32",
+            }
+        )
+
+        if not pa_version_under13p0:
+            # TODO: is this astype safe?
+            expected["DateCol"] = expected["DateCol"].astype("datetime64[us]")
```

Comment:
I'm not sure this is the appropriate way to cast from nanosecond precision to microsecond precision but it works. cc @jbrockmendel if you know a better way

Reply:
this is fine.  going from ns to us could silently round but is otherwise safe.  to raise rather than round use `ser.dt.as_unit("us", round_ok=False)`